pczt/src/roles/verifier/mod.rs

```diff
@@ -0,0 +1,32 @@
+use crate::Pczt;
+
+#[cfg(feature = "orchard")]
+mod orchard;
+#[cfg(feature = "orchard")]
+pub use orchard::OrchardError;
+
+#[cfg(feature = "sapling")]
+mod sapling;
+#[cfg(feature = "sapling")]
+pub use sapling::SaplingError;
+
+#[cfg(feature = "transparent")]
+mod transparent;
+#[cfg(feature = "transparent")]
+pub use transparent::TransparentError;
+
+pub struct Verifier {
+    pczt: Pczt,
+}
+
+impl Verifier {
+    /// Instantiates the Verifier role with the given PCZT.
+    pub fn new(pczt: Pczt) -> Self {
```

Comment:
I'm surprised that this takes ownership of the value, instead of just taking a reference. Is this just for symmetry with how the other roles are implemented?

Reply:
If it takes ownership, you can no longer use a value that has failed to verify, which seems desirable. (You could copy the value beforehand, but it still seems as though this API reduces the chance of making a class of errors.)

Reply:
The main reason is that the `into_parsed` methods consume the unparsed bundles, so if we took a reference here then we'd be internally cloning (and the caller can instead externally clone). The "lose access to the PCZT on detection of error" is a nice side-effect.